osu.Game.Rulesets.Taiko/Mods/TaikoModSingleTap.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Localisation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
+using osu.Game.Utils;
+using osu.Game.Rulesets.Taiko.UI;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public partial class TaikoModSingleTap : Mod, IApplicableToDrawableRuleset<TaikoHitObject>, IUpdatableByPlayfield
+    {
+        public override string Name => @"Single Tap";
+        public override string Acronym => @"SG";
+        public override LocalisableString Description => @"One key for dons, one key for kats.";
+
+        protected bool CheckValidNewAction(TaikoAction action)
+        {
+            if (action == TaikoAction.LeftCentre || action == TaikoAction.RightCentre)
+                return LastAcceptedDonAction == null || LastAcceptedDonAction == action;
+            if (action == TaikoAction.LeftRim || action == TaikoAction.RightRim)
+                return LastAcceptedKatAction == null || LastAcceptedKatAction == action;
+
+            return true;
+        }
```

Comment:
Please inline this into `checkCorrectAction()` for now. The fact that this is split and has the same checks duplicated that that method does is making this some very arduous and confusing reading.